pkg/gadgets/trace/open/tracer/bpf/opensnoop.h

```diff
@@ -24,6 +24,7 @@ struct event {
 	__u16 mode;
 	__u8 comm[TASK_COMM_LEN];
 	__u8 fname[NAME_MAX];
+	__u8 full_fname[NAME_MAX];
```

Comment:
I think I understand what you mean in general, but I am not sure about the practice. Do you mean that if the last member of the C struct is the `full_name`, since it is a null-terminated string it can be truncated in the ring-buffer? Is this working with giving in the last argument `bpf_perf_event_output` less than `sizeof(struct event)`?

Reply:
The last argument to `bpf_perf_event_output` should be a variable that takes the return value of `bpf_probe_read_user_str` into consideration. The [trace exec](https://github.com/inspektor-gadget/inspektor-gadget/blob/main/pkg/gadgets/trace/exec/tracer/bpf/execsnoop.bpf.c) gadget has a more complex example of this as in that case multiple strings are saved at the end.

Reply:
@mauriciovasquezbernal check out the code I have pushed, I have changed `full_fname` to be at PATH_MAX size and also putting variable size (depending on the string size) on the ring buffer.